src/commands/api-key/generate.ts

```diff
@@ -41,23 +45,69 @@ export default class GenerateCmd extends Command {
 		This key can be used to log into the CLI using 'balena login --token <key>',
 		or to authenticate requests to the API with an 'Authorization: Bearer <key>' header.
 `;
-	public static examples = ['$ balena api-key generate "Jenkins Key"'];
+	public static examples = [
+		'$ balena api-key generate "Jenkins Key"',
+		'$ balena api-key generate "Jenkins Key" 2025-10-30',
+		'$ balena api-key generate "Jenkins Key" never',
+	];
 
 	public static args = {
 		name: Args.string({
 			description: 'the API key name',
 			required: true,
 		}),
+		expiryDate: Args.string({
+			description: 'the expiry date of the API key, or "never" for no expiry',
+		}),
 	};
 
 	public static authenticated = true;
 
 	public async run() {
 		const { args: params } = await this.parse(GenerateCmd);
 
+		let expiryDateResponse: string | number = params.expiryDate ?? '';
```

Comment:
I would prefer undefined instead of empty string.
```suggestion
		let expiryDateResponse: string | number | undefined = params.expiryDate;
```
You can then also change the `if (params.expiryDate === undefined)` to use the new `expiryDateResponse` var, and then TS should be able to understand that `expiryDateResponse` can't be undefined after the `if () {}`.